lib/matplotlib/lines.py

```diff
@@ -790,8 +790,21 @@ def draw(self, renderer):
                     cap = self._dashcapstyle
                     join = self._dashjoinstyle
                 else:
+                    if tpath.should_simplify and affine.has_inverse:
+                        # the path simplification code expects the path
+                        # to be in pixel space, not data space, but the
+                        # renderer downstream expects the path to be
+                        # in data space. The transformation `affine`
+                        # will put the path into pixel space, and
+                        # affine.inverted() will transform it back to
+                        # data space. So, transform using affine, simplify
+                        # the path, and transform using the affine inverted.
+                        tpath = affine.inverted().transform_path(
+                            affine.transform_path(tpath).cleaned(simplify=True)
+                        )
```

Comment:
I really meant something like "Check if affine is used anywhere after.  If not, do /previous message/.  If it is, set a flag somewhere, and only use an identity affine when drawing the path".
Well perhaps the second one *would be* an premature optimization if needed, but please give the first one a try if possible, given that that would be fairly simple.